File: Python3.x/35-Search Insert Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for i, v in enumerate(nums):
            if v >= target: return i
        return len(nums)  # 最后一位插入

File: Algorithms/C++/35-Search_Insert_Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1, mid;
        while (l <= r) {
            mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] > target) {
                r = mid - 1;
            } else { // nums[mid] < target
                l = mid + 1;
            }
        }
        return l;
    }
};
